datafusion/physical-expr/src/utils.rs

```diff
@@ -235,6 +234,82 @@ pub fn ordering_satisfy_concrete<F: FnOnce() -> EquivalenceProperties>(
     }
 }
 
+#[derive(Clone, Debug)]
+pub struct ExprTreeNode<T> {
```

Comment:
Can you please add some documentation to this structure explaining its use?

Reply:
Sure, we will 👍

Reply:
Yes, please add some comments, specifically for the `data` member.

Reply:
This comment still seems to be unaddressed. Perhaps we can handle it in a follow on PR? This one is already pretty large and been outstanding for a while

Reply:
Took a note, we will add comments here in the next follow-on PR. Seems like it slipped through the cracks this time.